### DISCUSSION TITLE
Combining all masks to one mask?

### DISCUSSION THREAD

**Discussion by mrmoonshine24** [facebookresearch/detectron2#3882] [Q&A]
Hi all,

I am currently working on a schoolproject and therefor we need to use detectron2. However I want to combine all the detected masks from detectron2 in 1 masks, so basically having the following: [768, 1159, 3 ] where the 3 needs to be combined in 1. below you can see the code:

`mask_array = result['instances'].pred_masks.numpy()
print(mask_array.shape)
[5, 768, 1159]
 
num_instances = mask_array.shape[0]
mask_array = np.moveaxis(mask_array, 0, -1)
mask_array_instance = []
output = np.zeros_like(images) #black
#print('output',output)
for i in range(num_instances):
    mask_array_instance.append(mask_array[:, :, i:(i+1)])
    output = np.where(mask_array_instance[i] == True, 255, output)
masks = output      #mask
print(masks)
[768, 1159, 3] `
so the 3 needs to be just 1 mask...

thank you so much for your time and effort.

**Comment by alexander-soare** [comment]
when you get your mask array with shape [5, 768, 1159] just do `mask_array = np.any(mask_array, axis=0)`.

PS your for loop looks weird. You are reassigning the output each iteration.

**Reply by mrmoonshine24** [reply]
thanks for your replay, however I do need to keep the 1. when i am using the line of code you provided I will only have left the w , h of the picture. The mask is not there anymore or is it just not stated.. 

And the loop, I am still a early beginner so I may have borrowed someone else his code ... ;)

**Reply by alexander-soare** [reply]
If you want it to be (1, H, W) you can use the kwarg keeepdims=True in np any. 90% certain of that, but not at a computer to check
